sql/src/main/java/io/crate/action/sql/TransportBaseSQLAction.java

```diff
@@ -435,4 +301,101 @@ public void enable() {
     public void disable() {
         disabled = true;
     }
+
+
+    private static class StatsTableListener<TResponse extends SQLBaseResponse> implements ActionListener<TResponse> {
```

Comment:
probably call it *wrapper* and alos the listener delegate? same for killandretry